test/integration_test.js

```diff
@@ -648,6 +648,80 @@ describe("Integration tests", function() {
+              .then(newRecord => {
+                expect(newRecord.data.last_modified).to.exist;
+                return tasks.api.bucket("default").collection("tasks").updateRecord(
+                  {id: conflictingId, title: "remotely changed title"}, {patch: true});
+              })
+              .then(() => tasks.sync())
+              .then(syncResult => {
+                expect(syncResult.ok).eql(false);
+                expect(syncResult.conflicts).to.have.length.of(1);
+                // always pick our version, which has an older last_modified
+                return tasks.resolve(syncResult.conflicts[0], syncResult.conflicts[0].local);
+              })
+              .then(() => tasks.sync())
+              .then(syncResult => {
+                expect(syncResult.ok).eql(true);
+                expect(syncResult.conflicts).to.have.length.of(0);
+                expect(syncResult.updated).to.have.length.of(0);
+              })
+              .then(() => tasks.get(conflictingId))
+              .then(record => {
+                expect(record.data.title).eql("locally changed title");
+                expect(record.data._status).eql("synced");
+              });
+          });
+
+          it("should not skip other conflicts", () => {
+            const conflictingId2 = uuid4();
+            const rawCollection = tasks.api.bucket("default").collection("tasks");
+            return tasks.create({id: conflictingId2, title: "second title"},
+                                {useRecordId: true})
+              .then(() => tasks.sync())
+              .then(() =>
+                rawCollection.updateRecord(
+                  {id: conflictingId, title: "remotely changed title"}, {patch: true})
+                   )
+              .then(() =>
+                rawCollection.updateRecord(
+                  {id: conflictingId2, title: "remotely changed title2"}, {patch: true}))
+              .then(() => tasks.update({id: conflictingId, title: "locally changed title"},
+                                       {patch: true}))
+              .then(() => tasks.update({id: conflictingId2, title: "local title2"},
+                                       {patch: true}))
+              .then(() => tasks.sync())
+              .then(syncResult => {
+                expect(syncResult.ok).eql(false);
+                expect(syncResult.conflicts).to.have.length.of(2);
+                // resolve just one conflict and ensure that the other
+                // one continues preventing the sync, even though it
+                // happened "after" the first conflict
+                return tasks.resolve(syncResult.conflicts[1], syncResult.conflicts[1].local);
+              })
+              .then(() => tasks.sync())
+              .then(syncResult => {
+                expect(syncResult.ok).eql(false);
+                expect(syncResult.conflicts).to.have.length.of(1);
+                expect(syncResult.updated).to.have.length.of(0);
```

Comment:
we should be able to expect `published` to contain the change made locally (i.e. the resolution)